bundle/Templating/Twig/Extension/ContentViewRuntime.php

```diff
@@ -0,0 +1,212 @@
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Twig extension runtime for content rendering (view).
+ */
+class ContentViewRuntime
+{
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var \Symfony\Component\HttpKernel\Controller\ControllerResolverInterface
+     */
+    private $controllerResolver;
+
+    /**
+     * @var \Symfony\Component\HttpKernel\Controller\ArgumentResolverInterface
+     */
+    private $argumentResolver;
+
+    /**
+     * @var \Netgen\Bundle\EzPlatformSiteApiBundle\View\Builder\ContentViewBuilder
+     */
+    private $viewBuilder;
+
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\View\Renderer
+     */
+    private $viewRenderer;
+
+    /**
+     * @var \Psr\Log\LoggerInterface|\Psr\Log\NullLogger|null
+     */
+    private $logger;
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     * @param \Symfony\Component\HttpKernel\Controller\ControllerResolverInterface $controllerResolver
+     * @param \Symfony\Component\HttpKernel\Controller\ArgumentResolverInterface $argumentResolver
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\View\Builder\ContentViewBuilder $viewBuilder
+     * @param \eZ\Publish\Core\MVC\Symfony\View\Renderer $viewRenderer
+     * @param \Psr\Log\LoggerInterface|null $logger
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        ControllerResolverInterface $controllerResolver,
+        ArgumentResolverInterface $argumentResolver,
+        ContentViewBuilder $viewBuilder,
+        Renderer $viewRenderer,
+        ?LoggerInterface $logger = null
+    ) {
+        $this->requestStack = $requestStack;
+        $this->controllerResolver = $controllerResolver;
+        $this->argumentResolver = $argumentResolver;
+        $this->viewBuilder = $viewBuilder;
+        $this->viewRenderer = $viewRenderer;
+        $this->logger = $logger ?? new NullLogger();
+    }
+
+    /**
+     * Renders the HTML for a given $content.
+     *
+     * @param string $viewType
+     * @param \eZ\Publish\API\Repository\Values\ValueObject $contentOrLocation
+     * @param array $parameters
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentType
+     * @throws \eZ\Publish\Core\Base\Exceptions\UnauthorizedException
+     *
+     * @return string The HTML markup
+     */
+    public function renderContentView(
+        string $viewType,
+        ValueObject $contentOrLocation,
+        array $parameters = []
+    ): string {
+        $this->logger->critical('Twig function "ng_view_content" is experimental. Please report any issues on https://github.com/netgen/ezplatform-site-api/issues');
+
+        $content = $this->getContent($contentOrLocation);
+
+        $view = $this->viewBuilder->buildView([
+            'viewType' => $viewType,
```

Comment:
This might not be a good idea. These will end up on Sentry in production mode. A comment in phpdoc might be enough.

Reply:
Went on the wrong line, this was supposed to comment a critical logger call.

Reply:
Hm, do you mean about throwing `LogicException` below?

Reply:
OK, updated as suggested in https://github.com/netgen/ezplatform-site-api/pull/117/commits/4b98a242dcfb3a2b92f967514c44ae7edf473155